File: modules/EnchantmentsEnhance-Plugin/src/main/kotlin/org/pixeltime/enchantmentsenhance/manager/DatabaseManager.java
package org.pixeltime.enchantmentsenhance.manager;

import com.sk89q.worldedit.entity.Player;
import org.pixeltime.enchantmentsenhance.model.PlayerStat;
import org.pixeltime.enchantmentsenhance.mysql.DataStorage;

import java.util.ArrayList;

public class DatabaseManager {
    private static ArrayList<PlayerStat> playerStats = new ArrayList<>();


    public void addPlayerStat(Player player) {
        DatabaseManager.removePlayerStat(player.getName());
        PlayerStat playerStat = new PlayerStat();
        playerStat.setPlayername(player.getName());
        DataStorage.get().loadStats(playerStat);
        playerStats.add(playerStat);
    }


    public static PlayerStat getPlayerStat(final String playername) {
        for (final PlayerStat pData : playerStats) {
            if (pData.getPlayername().equals(playername)) {
                return pData;
            }
        }
        return null;
    }

    public static void removePlayerStat(String playername) {
        PlayerStat ps = getPlayerStat(playername);
        if (ps != null) {
            playerStats.remove(ps);
        }
    }

}

File: modules/EnchantmentsEnhance-Plugin/src/main/kotlin/org/pixeltime/enchantmentsenhance/manager/DatabaseManager.kt
package org.pixeltime.enchantmentsenhance.manager

import com.sk89q.worldedit.entity.Player
import org.pixeltime.enchantmentsenhance.model.PlayerStat
import org.pixeltime.enchantmentsenhance.mysql.DataStorage
import java.util.*

class DatabaseManager {


    fun addPlayerStat(player: Player) {
        DatabaseManager.removePlayerStat(player.name)
        val playerStat = PlayerStat()
        playerStat.playername = player.name
        DataStorage.get().loadStats(playerStat)
        playerStats.add(playerStat)
    }

    companion object {
        private val playerStats = ArrayList<PlayerStat>()

        fun getPlayerStat(playername: String): PlayerStat? {
            for (pData in playerStats) {
                if (pData.playername == playername) {
                    return pData
                }
            }
            return null
        }

        fun removePlayerStat(playername: String) {
            val ps = getPlayerStat(playername)
            if (ps != null) {
                playerStats.remove(ps)
            }
        }
    }

}
